applications/DeepSpeed-Chat/train.py

```diff
@@ -97,10 +87,10 @@ def parse_args():
         help="Directory for output of each step",
     )
     parser.add_argument(
-        "--num-gpus",
-        type=int,
-        default=1,
-        choices=(1, 8, 64),
+        "--deployment-type",
+        type=str,
+        default="single_gpu",
+        choices=("single_gpu", "single_node", "multi_node"),
```

Comment:
This could still bite us since there are many environments where: 
1. single_node != 8 gpus
2. multi_node != 64 gpus

Reply:
The single_node does not need to be on 8 GPUs. The way we launch it, deepspeed will simply use all available GPUs:
https://github.com/microsoft/DeepSpeedExamples/blob/8e4ec02c1545f7bd87d3bfe5daaafa5a5f1fe6a6/applications/DeepSpeed-Chat/training/step3_rlhf_finetuning/training_scripts/single_node/run_1.3b.sh#L27

Just tested locally (with 2GPUS) and using `single_node` deployment type does the correct thing.

Reply:
Good to know that DS will use all available GPUs. Another part of my concern is whether there is an expectation to total GPU memory for these scenarios. For example, does the single_node script assume 8*40GB available to fit the model using zero3?